### PR TITLE
Feature/registration ui fix

### PR DESCRIPTION
**Pull request by ykobuk**
Registration user fix

### CODE REVIEW

File: client/src/components/Nav/Registration.jsx [hunk 1]
```diff
@@ -32,43 +34,68 @@ class Registration extends Component {
...
              this.setState({msg: data.msg})
-             window.Materialize.toast(this.state.msg, 3000)
-             setTimeout(this.props.history.push('/'), 3000)
+             if(
+                 this.state.msg !== "We sent token to your email address"
+             ) {
+                 window.Materialize.toast(this.state.msg, 3000)
+                 setTimeout(this.props.history.push('/registration'), 3000)
+                 return null
+             }else{
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: Please add spaces `} else {` for better readable.

File: client/src/components/Nav/Registration.jsx [hunk 2]
```diff
@@ -32,43 +34,68 @@ class Registration extends Component {
         request('/registration',"POST", JSON.stringify(register_data))
             .then(data=>{
              this.setState({msg: data.msg})
-             window.Materialize.toast(this.state.msg, 3000)
-             setTimeout(this.props.history.push('/'), 3000)
+             if(
+                 this.state.msg !== "We sent token to your email address"
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: I understand it but seems really strange, better to send status: true/false from a server and handle it here.

File: client/src/components/Nav/Registration.jsx [hunk 3]
```diff
@@ -18,6 +19,7 @@ class Registration extends Component {
     register = (e) => {
         let register_data = {
```
Comment (RayMet) [inline client/src/components/Nav/Registration.jsx]: name of "register_data" must be something like that "registerData"

**Review by Merteg — CHANGES_REQUESTED** [review]
You didn't add a red underline to inputs when the input data is invalid.

**Review by RayMet — CHANGES_REQUESTED** [review]

**Review by RayMet — APPROVED** [review]

File: client/src/components/ConfirmEmail.jsx [hunk 4]
```diff
@@ -1,20 +1,39 @@
...
+        message_to_user: ''
     }
 
     componentDidMount(){
         request('/email_confirm/'+this.props.match.params.token)
             .then(data=>{
                 this.setState({msg: data.msg})
-                window.Materialize.toast(this.state.msg, 3000)
+                if (
+                    this.state.msg === "Your email address is confirmed"
```
Comment (Merteg) [inline client/src/components/ConfirmEmail.jsx:16]: I still think better to send boolean status from a server.

File: client/src/components/ConfirmEmail.jsx [hunk 5]
```diff
@@ -1,20 +1,39 @@
...
+                ){
+                    this.setState({message_to_user:
+                                  "Your email address is confirmed, please press the Home button and Log in"})
```
Comment (Merteg) [inline client/src/components/ConfirmEmail.jsx:19]: Wrong identation

File: client/src/components/Nav/Registration.jsx [hunk 6]
```diff
@@ -16,59 +20,97 @@ class Registration extends Component {
...
+                 this.state.msg === "Your email address is already registered"
+             ) {
+                 this.setState({check_email: 'invalid email'})
+                 window.Materialize.toast(this.state.msg, 3000)
+                 setTimeout(this.props.history.push('/registration'), 3000)
+                 return null
+             } else if (
+                 this.state.msg === "Your nickname is taken, please choose another"
+             ) {
+                 this.setState({check_nickname: 'invalid nickname'})
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx:51]: Better to start phrase from uppercase letter

File: client/src/components/Nav/index.jsx [hunk 7]
```diff
@@ -8,10 +8,10 @@ import { isLogged } from '../../utils'
...
 
 class Nav extends Component {
-  constructor(props) {
-    super(props);
-    this.update = this.update.bind(this);
-  }
+    constructor(props) {
+        super(props);
+        this.update = this.update.bind(this);
+    }
```
Comment (Merteg) [inline client/src/components/Nav/index.jsx:14]: Why did you add spaces?

**Review by yazhyvyi — APPROVED** [review]

**Pull request merged**